Update Foundry and CI Configs, Run Linter

apbendi opened the pull request.

apbendi committed:
Update foundry and ci configs

apbendi committed:
Run linter

mds1 commented on .github/workflows/ci.yml:
```diff
@@ -63,7 +63,15 @@ jobs:
+          token: ${{ secrets.GITHUB_TOKEN }}
 
       - name: Check formatting
         run: scopelint check
```
Just as a sanity check on the version being used in case there's any issues
```suggestion
        run: |
          scopelint --version
          scopelint check
```

apbendi replied:
added in 29b3bedc50644c79b9f947fe63322c7a78faefd3

mds1 commented on .github/workflows/ci.yml:
```diff
@@ -63,7 +63,15 @@ jobs:
 
       - name: Install scopelint
-        run: cargo install scopelint
+        uses: engineerd/configurator@v0.0.8
```
_(arbitrary comment location to start a thread)_

I don't see the `coverage` step added here (it was on a branch until now, just merged). Maybe you left it off intentionally which is fine, just calling that out in case you want to add it

apbendi replied:
Ahh yeah I pulled the updates from `main` and it wasn't added yet. Now include in 29b3bedc50644c79b9f947fe63322c7a78faefd3

apbendi committed:
Add coverage and version check to ci

apbendi committed:
Reduce coverage threshold

Because we implement several methods in the Governor soley to disambiguate
between ancestor class implementations, our coverage is calculated as very
low. For now, we'll simply reduce the required threshold—this will give us
a report from CI without causing failures.
